### PR TITLE
feat: add unwrapped token address to linearPools

### PR DESCRIPTION
**Pull request by evgenyboxer**
# Description

Adds "unwrappedTokenAddress" to linearPool data.

Fixes # (issue)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Other

## How should this be tested?

Please provide instructions so we can test. Please also list any relevant details for your test configuration.

- [ ] `unwrappedTokenAddress` should be added to the onchain linearPool data

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have commented my code where relevant, particularly in hard-to-understand areas
- [ ] My changes generate no new console warnings
- [ ] The base of this PR is `master` if hotfix, `develop` if not

### CODE REVIEW

File: src/services/balancer/contracts/contracts/vault.ts [hunk 1]
```diff
@@ -112,17 +112,34 @@ export default class Vault {
...
+        poolMulticaller.call(
+          `${address}.unwrappedTokenAddress`,
+          wrappedToken,
+          'ATOKEN'
+        );
       });
+
+      const callResult = await poolMulticaller.execute();
+      if (callResult != null) {
+        merge(result.linearPools, callResult);
```
Comment (evgenyboxer) [inline src/services/balancer/contracts/contracts/vault.ts:141]: @garethfuller WDYT about this way of merging the results? They both share the same `key` (the address), so I think its safe to do it.

**Comment by evgenyboxer** [comment]
I've named the ABI file: `StaticATokenLM.json` as the name of the contract appears on etherscan

File: src/services/balancer/contracts/contracts/vault.ts (hunk 1 above)
Comment (garethfuller) [reply src/services/balancer/contracts/contracts/vault.ts:141]: Can you not just modify the `result` like we're doing elsewhere? I think execute already handles this kind of merging of data.
```ts
Object.entries(wrappedTokensMap).forEach(([address, wrappedToken]) => {
  poolMulticaller.call(
    `linearPools.${address}.unwrappedTokenAddress`,
    wrappedToken,
    'ATOKEN'
  );
});

const result = await poolMulticaller.execute(result);
```

**Review by garethfuller — COMMENTED** [review]
Looks good, just one other thing, in [usePoolQuery](https://github.com/balancer-labs/frontend-v2/blob/feat/evgeny/add-unwrapped-tokens/src/composables/queries/usePoolQuery.ts#L95-L100) could you add an `unwrappedTokens` array too.

File: src/services/balancer/contracts/contracts/vault.ts (hunk 1 above)
Comment (evgenyboxer) [reply src/services/balancer/contracts/contracts/vault.ts]: Thanks, much cleaner :)

File: src/composables/queries/usePoolQuery.ts [hunk 2]
```diff
@@ -157,8 +159,20 @@ export default function usePoolQuery(
...
 
-    console.log('pool', { onchain: onchainData, ...decoratedPool });
-    return { onchain: onchainData, ...decoratedPool };
+    let unwrappedTokens: Pool['unwrappedTokens'];
+
+    if (isStablePhantomPool && onchainData.linearPools != null) {
+      unwrappedTokens = Object.entries(onchainData.linearPools).map(
+        ([, linearPool]) => linearPool.unwrappedTokenAddress
+      );
+    }
```
Comment (garethfuller) [inline src/composables/queries/usePoolQuery.ts:168]: We're already modifying the pool object with two new arrays of pool tokens in `getLinearPoolAttrs` could you not just move the additional construction of unwrappedTokens here too: https://github.com/balancer-labs/frontend-v2/blob/0f3c9c1569df580b109e4d1563ffbdc7faa4ea5b/src/composables/queries/usePoolQuery.ts#L95-L100

Comment (evgenyboxer) [reply src/composables/queries/usePoolQuery.ts:168]: Right, but `onchainData ` (where the `unwrappedTokenAddress` is) is computed after the `getLinearPoolAttrs` method. (so it doesn't exist there yet).

So you mean we want to compute `onchainData` before calling `getLinearPoolAttrs` right?

Comment (garethfuller) [reply src/composables/queries/usePoolQuery.ts:168]: Ahh crap, sorry! You're right. Would be good if we could get these tokens from the subgraph call to the linear pools in `getLinearPoolAttrs` but I guess we can't. We'll have to clean all this up later.

Comment (evgenyboxer) [reply src/composables/queries/usePoolQuery.ts:168]: Yeah I agree.

Are you happy for this to be merged for now?

**Review by garethfuller — APPROVED** [review]

**Pull request merged**